main.go

```diff
@@ -30,12 +30,28 @@ func HandleLambdaEvent() error {
 		session)
 }
 
+func retry(maxAttempts int, f func() error) (int, error) {
```

Comment:
As discussed. Not sure if this is a Golang limitation but this seems like it's returning an int (attempt count) just to make the test work rather than because it's useful in the implementation. 

Also means that technically we're just testing a proxy for the method call count, rather than the actual thing we care about. i.e this would make the test pass but be wrong:

```go
func retry(maxAttempts int, f func() error) (int, error) {
    var err error
    var attempts int

    err = f()
    return 4, err
}
```